pkg/database/database.go

```diff
@@ -271,16 +270,46 @@ func callbackIncrementMetric(ctx context.Context, m *stats.Int64Measure, table s
 		// Add realm so that metrics are groupable on a per-realm basis.
 		field, ok := scope.FieldByName("realm_id")
 		if ok && field.Field.CanInterface() && field.Field.Interface() != nil {
-			realmID := field.Field.Interface()
-
-			var err error
-			ctx, err = tag.New(ctx, tag.Upsert(observability.RealmTagKey, fmt.Sprintf("%d", realmID)))
-			if err != nil {
-				_ = scope.Err(fmt.Errorf("failed to add realm tag to metrics: %w", err))
+			realmIDRaw := field.Field.Interface()
+
+			var realmID uint
+			switch t := realmIDRaw.(type) {
+			case uint:
+				realmID = t
+			case uint8:
```

Comment:
slightly more succinct is:

```
case uint8, uint16, uint32, ... :
  realmId = uint(t)
```

Reply:
@jeremyfaller that does not work - maybe you know a good Go compiler engineer who could fix it, but `t` is only the concrete type when given a single element in the case statement. Otherwise t is `interface{}`. I would love to do the more succinct version 😄